.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,24 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          cache: false
+      - name: Golangci-lint
+        uses: golangci/golangci-lint-action@v6.0.1    
+        with:
+          args: --timeout 3m
```

Comment:
plz set linters u wanna run and configure it
we use such linters:
   - megacheck
    - govet
    - unconvert
    - megacheck
    - gas
    - gocyclo
    - dupl
    - misspell
    - unparam
    - typecheck
    - ineffassign
    - stylecheck
    - gochecknoinits
    - exportloopref
    - gocritic
    - revive
    - nakedret
    - gosimple
    - prealloc
    - unused
    - dogsled
    - gocognit
    - gofumpt
    - noctx
    - testpackage
    - lll

and this is configuration:
```yaml
  govet: 
    # in addition to default (see golangci docs for list of defaults)
    enable:
      - shadow
  gocyclo:
    min-complexity: 25
  dupl:
    threshold: 100
  goconst:
    min-len: 2
    min-occurrences: 2
  misspell:
    locale: US
  gocritic:
    enabled-tags:
      - performance
      - style
      - experimental
    disabled-checks:
      - wrapperFunc
      - hugeParam
      - rangeValCopy
  lll:
    line-length: 140
```